wl/utils/Utils.java

```diff
@@ -41,20 +41,15 @@ public ChecksummedFile(File f) {
 
 		private String cs() { return file.isFile() ? checksum(file) : ""; }
 
-		public boolean valid() { return cs().equals(cachedChecksum); }
-
-		@Override
-		public int compareTo(Object o) {
-			return (o instanceof ChecksummedFile && valid() && ((ChecksummedFile)o).valid()) ?
-                file.compareTo(((ChecksummedFile)o).file) :
-                -1;
```

Comment:
The problem is probably the constant -1. Instead it should be something like `cachedChecksum.compareTo(o.cachedChecksum)`